cmd/clusterctl/client/repository/template_test.go

```diff
@@ -84,34 +86,24 @@ func Test_newTemplate(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
 			got, err := NewTemplate(tt.args.rawYaml, tt.args.configVariablesClient, tt.args.targetNamespace, tt.args.listVariablesOnly)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got.Variables(), tt.want.variables) {
-				t.Errorf("got.Variables() = %v, want = %v ", got.Variables(), tt.want.variables)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
```

Comment:
nit we should be consistent in having assertion on the operation err/on wantErr like in Test_componentsClient_Get